### PR TITLE
fix(vlm): forward image payloads in async rollouts

### PR DESCRIPTION
**Pull request by jinglinglingling**
# What does this PR do ?

Preserves native multimodal side-channel fields when SingleController dispatches async VLM rollouts, so vLLM and the training forward receive the same image input.

The async per-sample path previously rebuilt generation input from tokenized message data but dropped fields such as `vllm_content` and `vllm_images`. This made vLLM generation image-blind while Megatron recomputed log probabilities with the image, producing large `token_mult_prob_error` values.

For multi-turn rollouts, later turns retain native media while clearing stale `vllm_content`, allowing the updated tokenized prefix to be used.

This PR does not modify recipes or generation lengths. The existing defaults remain Qwen `max_new_tokens=1024` and Nemotron `max_new_tokens=4096`.

Validation results:
- Controlled unfixed Qwen reproduction: `train/token_mult_prob_error = 2.77639`
- Same controlled setup after the fix, vLLM TP1/TP2/TP4: approximately `1.02`
- Native Qwen configuration (`max_new_tokens=1024`, 8 prompts x 16 generations), 100/100 steps: all TMPE values below `1.05`; range `1.02006–1.02614`; truncation rate `0`
- Native Qwen W&B run: https://wandb.ai/nvidia/rohit-async_grpo_v2/runs/4t6osucc
- Native Nemotron configuration (`max_new_tokens=4096`, 8 prompts x 16 generations), 100/100 steps completed. Most aggregate TMPE values were approximately `1.03`; three outliers remain under investigation, including one immediately after checkpoint resume.
- Native Nemotron W&B runs: https://wandb.ai/nvidia/rohit-async_grpo_v2/runs/5g2qyqmb and https://wandb.ai/nvidia/rohit-async_grpo_v2/runs/y2govxqy

## Reproduction commands

Qwen native-length 100-step validation:

```bash
uv run --locked examples/run_grpo_single_controller.py \
  --config examples/configs/recipes/vlm/vlm_grpo-qwen2.5-vl-3b-instruct-clevr-1n8g-megatron-single-controller-async.v1.yaml \
  grpo.max_num_steps=100 \
  logger.wandb_enabled=true \
  logger.tensorboard_enabled=true \
  logger.monitor_gpus=false \
  logger.wandb.project=rohit-async_grpo_v2 \
  logger.wandb.name=linglinj-qwen-image-fix-native1024
```

Nemotron native-length 100-step validation (3 nodes / 8 GPUs per node):

```bash
uv run --locked examples/run_grpo_single_controller.py \
  --config examples/configs/recipes/vlm/vlm_grpo-nemotron-omni-30ba3b-clevr-3n8g-megatron-tp8ep8-single-controller-async.v1.yaml \
  grpo.max_num_steps=100 \
  policy.train_global_batch_size=128 \
  logger.wandb_enabled=true \
  logger.tensorboard_enabled=true \
  logger.monitor_gpus=false \
  logger.wandb.project=rohit-async_grpo_v2 \
  logger.wandb.name=linglinj-nemotron-native4096-gbs128
```

### CODE REVIEW

**Comment by jinglinglingling** [comment]
ok to test

File: tests/unit/experience/test_rollout_manager.py [hunk 1]
```diff
@@ -147,13 +148,22 @@ async def generate_async(self, data):
     ]
 
     assistant_message, input_lengths, _ = _run(
-        manager._generate_response(message_log, ["<stop>"])
+        manager._generate_response(
```
Comment (rohitrango) [inline tests/unit/experience/test_rollout_manager.py:152]: [`tests/unit/experience/test_rollout_manager.py:151`](https://github.com/NVIDIA-NeMo/RL/blob/2bb044ee2918b392842bdb87c0051e2f07778e62/tests/unit/experience/test_rollout_manager.py#L151)

This calls `_generate_response` with already assembled native data, so it can pass even if the new [`_run_single_rollout` extraction](https://github.com/NVIDIA-NeMo/RL/blob/2bb044ee2918b392842bdb87c0051e2f07778e62/nemo_rl/experience/rollout_manager.py#L574) or [later-turn clearing](https://github.com/NVIDIA-NeMo/RL/blob/2bb044ee2918b392842bdb87c0051e2f07778e62/nemo_rl/experience/rollout_manager.py#L606) regresses. Could you add a two-turn `_run_single_rollout` test that checks original `vllm_content` plus media on turn 1 and `None` content plus unchanged image/audio/video on turn 2, as the [sibling rollout test](https://github.com/NVIDIA-NeMo/RL/blob/2bb044ee2918b392842bdb87c0051e2f07778e62/tests/unit/experience/test_rollouts.py#L768-L855) does?

File: nemo_rl/experience/rollout_manager.py [hunk 2]
```diff
@@ -570,6 +571,11 @@ async def _run_single_rollout(
     ) -> tuple[Completion, dict]:
         """Run one multi-turn rollout for a single generation index."""
         current_message_log = copy.deepcopy(input_sample["message_log"])
+        native_generation_data = {
```
Comment (rohitrango) [inline nemo_rl/experience/rollout_manager.py:577]: [`nemo_rl/experience/rollout_manager.py:574`](https://github.com/NVIDIA-NeMo/RL/blob/2bb044ee2918b392842bdb87c0051e2f07778e62/nemo_rl/experience/rollout_manager.py#L574)

This logic changes VLM generation inputs and the PR reports TMPE moving from `2.77639` to about `1.02`, but the current label is `CI:Lfast`. The [CI guide](https://github.com/NVIDIA-NeMo/RL/blob/2bb044ee2918b392842bdb87c0051e2f07778e62/.agents/contributor-skills/cicd/SKILL.md#L56-L64) reserves `CI:Lfast` for changes with no logic change and assigns convergence-affecting changes to `CI:L2`; could you switch this PR to `CI:L2` before merge?

Comment (rohitrango) [reply nemo_rl/experience/rollout_manager.py:577]: changed it

File: tests/unit/experience/test_rollout_manager.py (hunk 1 above)
Comment (jinglinglingling) [reply tests/unit/experience/test_rollout_manager.py:152]: Added a two-turn `_run_single_rollout` regression test in ec12bc64. It verifies that turn 1 receives the original `vllm_content` and all native media, while turn 2 receives `vllm_content=None` with the same image/audio/video objects.

**Comment by jinglinglingling** [comment]
Addressed the remaining review comment with a two-turn `_run_single_rollout` regression test in `ec12bc64`. The PR is labeled `CI:L2`; @rohitrango, could you approve the latest SHA with `/ok to test ec12bc64` so the L2 suite can run again?

**Comment by rohitrango** [comment]
could you sign your latest commit? once it passes DCO I will re-run

**Review by terrykong — APPROVED** [review]
